core/src/main/java/org/stellar/anchor/client/DefaultClientService.java

```diff
@@ -0,0 +1,86 @@
+package org.stellar.anchor.client;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.FilenameUtils;
+import org.stellar.anchor.api.exception.InvalidConfigException;
+import org.stellar.anchor.api.exception.SepNotFoundException;
+import org.stellar.anchor.config.ClientsConfig;
+import org.stellar.anchor.util.FileUtil;
+import org.stellar.anchor.util.GsonUtils;
+import org.yaml.snakeyaml.Yaml;
+
+public class DefaultClientService implements ClientService {
+  static final Gson gson = GsonUtils.getInstance();
+  List<CustodialClientConfig> custodialClients;
+  List<NonCustodialClientConfig> nonCustodialClients;
+
+  public static DefaultClientService fromClientsConfig(ClientsConfig clientsConfig)
+      throws InvalidConfigException {
+    return switch (clientsConfig.getType()) {
+      case FILE -> fromFile(clientsConfig.getValue());
+      case URL -> throw new InvalidConfigException("URL is not supported yet.");
+      default -> throw new InvalidConfigException(
+          String.format("clients.type:%s is not supported.", clientsConfig.getType()));
+    };
```

Comment:
We need to continue supporting in-line clients config. Ideally businesses wouldn't have to change their config files at all and continue using the `clients:` top-level attribute

Reply:
I thought this is why we are doing it for 3.0 because it's a breaking change? We do want to replace the in-line config with the separate file

Reply:
What should we do if businesses happened to have ClientsConfig specified both in-line and in a separate file? Should we prioritize one over the other?

Assets previously had an in-line definition feature, but it is not being utilized. Since ClientsConfig is being adopted by the business, shall we encourage them to use the separate file instead?

Reply:
We want to support both in-line and a separate file. In-line still makes sense if you're not using environment variables, since you can have all your config in one place.

Reply:
Just to confirm with you that by in-line you mean it can be defined in the default config as 
```
clients:
  - type:
  - value:
```

If so that's already included in this PR

Reply:
When I say inline, I mean specifying the clients list in the same file, not a separate one. This is supported for assets and we should support it for clients.

Reply:
I don't think assets supports the exact format you described. For assets it's either defined through file reference or yaml/json string 
```
assets:
  - type: file
  - value: file:assets.yaml
```
or
```
assets
  - type: yaml
  - value: "assets in yaml string not yaml format"
```

Reply:
I just tested and you're right, but this is a surprise to me. I thought the `yaml` type meant you could use yaml in the same file. I don't think anyone is going to provide a yaml-formatted or json-formatted string.

I think this is something we should support. It makes sense to support providing a file path to a yaml file when you're configuring everything else via envvars, but if you are using yaml-based config already, it doesn't make sense for the anchor platform to force you to put your assets (or clients) config in a different file. 

@philipliu @reecexlm @Ifropc what are your thoughts on this?

Reply:
I agree. It seems counterintuitive to force user to place your assets (or clients) configuration in a different file. 

> I don't think anyone is going to provide a yaml-formatted or json-formatted string
That's what I thought. This in-line support for assets is not being utilized at all. We might just drop it.

Reply:
Agree that we should do this. I think if we are going to allow inline client configuration, we should also allow inline asset configuration at the same time. This seems like an additive change, so let's implement it later to not block this PR.